Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs

```diff
@@ -0,0 +1,317 @@
+    /// <i>Does not work with the html element.</i>
+    /// </summary>
+    /// <param name="cssSelector">CSS selector of the control object.</param>
+    public void IgnoreCssSelector ([NotNull] string cssSelector)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty ("cssSelector", cssSelector);
+
+      ExcludedElements.Add (cssSelector);
+    }
+
+    /// <summary>
+    /// Accessibility rule to ignore during the accessibility analysis.
+    /// </summary>
+    /// <param name="rule">The <see cref="AccessibilityRuleID"/> to ignore when executing the analysis.</param>
+    public void IgnoreRule (AccessibilityRuleID rule)
+    {
+      ExcludedRules.Add (AccessibilityRuleIDConverter.ConvertToString (rule));
+    }
+
+    /// <summary>
+    /// Analyzes specific ControlObject on Webpage.
+    /// </summary>
+    /// <param name="controlObject">ControlObject to analyze.</param>
+    /// <returns> Results of the analysis.</returns>
+    [NotNull]
+    public AccessibilityResult Analyze ([NotNull] ControlObject controlObject)
+    {
+      ArgumentUtility.CheckNotNull ("controlObject", controlObject);
+
+      return Analyze ($"#{controlObject.GetHtmlID()}");
+    }
+
+    /// <summary>
+    /// Analyzes a specific element on webpage.
+    /// </summary>
+    /// <param name="cssSelector">CSS selector of the ControlObjects to analyze.</param>
+    /// <returns> Results of the analysis.</returns>
+    [NotNull]
+    public AccessibilityResult Analyze ([NotNull] string cssSelector)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty ("cssSelector", cssSelector);
+
+      return GetAccessibilityResult (cssSelector);
+    }
+
+    /// <summary>
+    /// Analyzes all content on a webpage.
+    /// Switches to the outermost frame before the analysis and back to
+    /// the original frame after the analysis completed.
+    /// </summary>
+    /// <returns> Results of the analysis.</returns>
+    [NotNull]
+    public AccessibilityResult Analyze ()
+    {
+      var outerFrame = (string) JsExecutor.ExecuteScript ("return self.name;");
+      if (outerFrame != "")
+        WebDriver.SwitchTo().DefaultContent();
+
+      var result = GetAccessibilityResult (null);
+      if (outerFrame != "")
+        WebDriver.SwitchTo().Frame (outerFrame);
+
+      return result;
+    }
+
+    private AccessibilityResult GetAccessibilityResult (string cssSelector)
+    {
+      var source = AxeSourceProvider.GetSource();
+
+      if (!AxeIsInjected())
+      {
+        using (new PerformanceTimer (Logger, "aXe has been injected."))
+        {
+          InjectAxeSource (source);
+        }
+      }
+
+      var axeRunFunctionCall = BuildAxeRunFunctionCall (cssSelector);
+
+      string result;
+      using (new PerformanceTimer (Logger, "Accessibility analysis has been performed."))
+      { 
+        result = (string) JsExecutor.ExecuteAsyncScript (axeRunFunctionCall) ?? GetRawAxeResultForInternetExplorer (axeRunFunctionCall);
```

Comment:
am not sure I want to use null coallese here. two non-primitive statements chained and such. what does the rest of the code say about this style?